cmd/influx_inspect/type_conflicts/schema.go

```diff
@@ -0,0 +1,149 @@
+package typecheck
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	errors2 "github.com/influxdata/influxdb/pkg/errors"
+)
+
+type UniqueField struct {
+	Database    string `json:"database"`
+	Retention   string `json:"retention"`
+	Measurement string `json:"measurement"`
+	Field       string `json:"field"`
+}
+
+type FieldTypes map[string]struct{}
+type Schema map[string]FieldTypes
+
+func (ft FieldTypes) MarshalText() (text []byte, err error) {
+	s := make([]string, 0, len(ft))
+	for f := range ft {
+		s = append(s, f)
+	}
+	return []byte(fmt.Sprintf("%s", strings.Join(s, ","))), nil
+}
+
+func (ft *FieldTypes) UnmarshalText(text []byte) error {
+	if *ft == nil {
+		*ft = make(FieldTypes)
+	}
+	for _, ty := range strings.Split(string(text), ",") {
+		(*ft)[ty] = struct{}{}
+	}
+	return nil
+}
+
+func NewSchema() Schema {
+	return make(Schema)
+}
+
+func SchemaFromFile(filename string) (Schema, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("unable to open schema file %q: %w", filename, err)
+	}
+
+	s := NewSchema()
+	if err := s.Decode(f); err != nil {
+		return nil, fmt.Errorf("unable to decode schema file %q: %w", filename, err)
+	}
+	return s, nil
+}
+
+func (uf *UniqueField) String() string {
+	return fmt.Sprintf("%s.%s.%s.%s", uf.Database, uf.Retention, uf.Measurement, uf.Field)
```

Comment:
Is a "." character permissible in any of those?  i.e. could we have a two different UniqueField's with the same string?  Appreciate it's very unlikely, so can probably just be ignored.

Reply:
If the schema name is quoted, `.` is permitted. If unquoted, it is not.  I added quotes to the output with `%q` formatting, which is less human-readable but should handle all cases.